ui/src/app/shared/components/tri-state-button.component.ts

```diff
@@ -0,0 +1,86 @@
+import {Component,Input,Output,OnInit, AfterViewInit, DoCheck,
+  ViewChild,EventEmitter,ChangeDetectionStrategy,
```

Comment:
EventEmitter, ChangeDetectionStrategy, Renderer, ElementRef, forwardRef Pipe, PipeTransform are not being used.